### PR TITLE
feat: Course Level Error Handling for Empty States

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Course Level Error Handling for Empty States
- Empty state for Home, Videos, Dates, and Discussion tabs.
- Empty state for handouts, and Announcements screen.

#### Tickets
- [LEARNER-10039](https://2u-internal.atlassian.net/browse/LEARNER-10039)
- [Issue# 339](https://github.com/openedx/openedx-app-android/issues/339)

| Home Tab | Video Tab | Dates Tab |
| ------------- | ------------- | ------------- |
|  |  |  |

| Discussion Tab | Handouts Screen | Announcements Screen |
| ------------- | ------------- | ------------- |
|  |  |  |

### CODE REVIEW

File: course/src/main/res/drawable/course_ic_no_announcements.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (omerhabib26) [reply course/src/main/res/drawable/course_ic_no_announcements.xml:1]: we can move these icons in core.

File: course/src/main/res/drawable/course_ic_no_videos.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (omerhabib26) [inline course/src/main/res/drawable/course_ic_no_videos.xml:1]: same here

File: discussion/src/main/java/org/openedx/discussion/presentation/topics/DiscussionTopicsScreen.kt [hunk 3]
```diff
@@ -253,6 +256,12 @@ private fun DiscussionTopicsUI(
                                 }
 
                                 DiscussionTopicsUIState.Loading -> {}
```
Comment (omerhabib26) [inline discussion/src/main/java/org/openedx/discussion/presentation/topics/DiscussionTopicsScreen.kt:258]: Can we add CircularProgress for loading?

**Review by omerhabib26 — CHANGES_REQUESTED** [review]
Some minor code improvements.

**Review by dixidroid — APPROVED** [review]

File: discussion/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -38,6 +38,7 @@
     <string name="discussion_unnamed_subcategory">Unnamed subcategory</string>
 
 
+
```
Comment (omerhabib26) [inline discussion/src/main/res/values/strings.xml:41]: unused code

**Review by omerhabib26 — APPROVED** [review]
only a minor nit.

**Comment by farhan-arshad-dev** [comment]
We're closing this PR and moving these changes to our own fork, as they are critical for us. After our release, we hope to find a way to contribute to these changes back, thanks!

**Pull request closed without merging**